lib/active_model/serializer/reflection.rb

```diff
@@ -119,6 +119,7 @@ def build_association(subject, parent_serializer_options)
               serializer_options(subject, parent_serializer_options, reflection_options)
             )
           rescue ActiveModel::Serializer::CollectionSerializer::NoSerializerError
+            warn "No serializer found for #{association_value}."
```

Comment:
Agreed, I always disliked the whole `build_reflection` flow and how it was written. Let's clean this up in a subsequent PR, as this one is simply about adding the warning.